lib/index.js

```diff
@@ -172,19 +146,31 @@ internals.parseServer = function (server, relativeTo) {
 
 internals.parsePlugin = function (plugin, relativeTo) {
 
-    plugin = Hoek.cloneWithShallow(plugin, ['options']);
-    if (typeof plugin === 'string') {
-        plugin = { register: plugin };
+    const pluginObject = { plugin };
+    if (typeof plugin.options === 'object') {
+        pluginObject.options = plugin.options;
     }
 
-    if (typeof plugin.register === 'string') {
+    if (typeof plugin === 'string') {
+        let path = plugin;
+        if (relativeTo && path[0] === '.') {
+            path = Path.join(relativeTo, path);
+        }
+
+        pluginObject.plugin = require(path);
+    }
+    else if (typeof plugin.register === 'string') {
         let path = plugin.register;
         if (relativeTo && path[0] === '.') {
             path = Path.join(relativeTo, path);
         }
 
-        plugin.register = require(path);
+        pluginObject.plugin = require(path);
```

Comment:
I don't understand this change - the required field is 'register' not 'plugin'.  I don't see any change regarding that in the hapi 17 docs.

Reply:
It's entirely possible that I'm misunderstanding, but [here is the change I'm looking at](https://github.com/hapijs/hapi/issues/3618).

>Switch plugins to use a simple object with a register property. All the other attributes move to the object alongside register.

and

>Change server.register() to support { plugin, options } instead of { register, options }


Additionally (and this may very well be an error on my side) when I try to register the plugins using
```js

exports.register = function (server, options) {

    server.expose('hello', options.who || 'world');
};

exports.register.attributes = {
    name: 'helloworld',
    multiple: false
};
```
I get the error:
```js
Error: Invalid plugin options {
  "plugin": {
    "register": function (server, options) {\n\n    server.expose('hello', options.who || 'world');\n},
    "name" [1]: -- missing --
  }
}

[1] "name" is required

```

Reply:
I misread the documentation.

I'm tempted to have glue only call server.register() a single time, passing all plugins at once.

In which case, the new schema would be:
```javascript
{
    options: Joi.object({
        relativeTo: Joi.string(),
        preRegister: Joi.func().allow(false)
    }),
    manifest: Joi.object({
        server: Joi.object(),
        register: Joi.Object({
           plugins: Joi.array().items(
             Joi.string(),
             Joi.object()
           ),
           options: Joi.any()
        ))
    })
}
```
example manifests:
```javascript
{register: {
  plugins: ["myplugin"]
}}
```
```javascript
{register: {
  plugins: [require("myplugin")],
  options: registerOptions
}}
```
```javascript
{register: {
  plugins: [{plugin: "myplugin", options: myPluginOptions}]
}}
```
```javascript
{register: {
  plugins: [
    {plugin: "myplugin", options: myPluginOptions},
    {plugin: "otherplugin", options: otherPluginOptions, routes: overrideRegisterRoutesOption},
    "thirdplugin",
    {plugin: require("bestplugin"), options: bestsPluginOptions}
  ],
  options: registerOptions
}}
```
And from that, if an item from the register.plugins array is a string we'll require() it, otherwise we'll pass the register.plugins array directly to server.register() without any additional validation, and let hapi itself validate the input.  As far as that goes, I'd be fine with simplifying the manifest  schema to be: 
```javascript
Joi.object({
  server: Joi.object(),
  register: Joi.Object({
    plugins: Joi.array(),
    options: Joi.any()
  ))
})
```
As we don't care what format the registration options are, and we don't care what the plugin format is, except if its a string, we require() it.  Let them pass whatever to hapi, where hapi is the source of truth to validate what is provided.

Reply:
I like that. I'll do that refactor on this PR this weekend :+1: